### ISSUE TITLE
[Bug]: 400 Error on vertex_ai/gemini-1.5-pro with nested Pydantic schema in response_format

### ISSUE THREAD

**Issue by pvachon-ivadolabs** [BerriAI/litellm#6027]
### What happened?

I was going through the tutorial on structured outputs and while this runs fine 
```python
class CalendarEvent(BaseModel):
      name: str
      date: str
      participants: list[str]

class EventsList(BaseModel):
    events: list[CalendarEvent]

response = litellm.completion(
    model="gpt-4o-2024-08-06",
    messages=messages,
    response_format=EventsList
)
```

The same call with `vertex_ai/gemini-1.5-pro` returns a 400
```python
class CalendarEvent(BaseModel):
      name: str
      date: str
      participants: list[str]

class EventsList(BaseModel):
    events: list[CalendarEvent]

response = litellm.completion(
    model="vertex_ai/gemini-1.5-pro",
    messages=messages,
    response_format=EventsList
)
```

I can however, get it to work by converting the pydantic objects like so
```python
response = litellm.completion(
    model="vertex_ai/gemini-1.5-pro",
    messages=messages,
    response_format={
        "type": "json_schema",
        "json_schema": {
            "schema": jsonref.replace_refs(EventsList.model_json_schema())["properties"]["events"],
        },
        "strict": True
    }
)
```

I would expect litellm to provide a consistent interface for structured outputs though.

### Relevant log output

```shell
---------------------------------------------------------------------------
HTTPStatusError                           Traceback (most recent call last)
File .venv/lib/python3.11/site-packages/litellm/llms/vertex_ai_and_google_ai_studio/gemini/vertex_and_google_ai_studio_gemini.py:1280, in VertexLLM.completion(self, model, messages, model_response, print_verbose, custom_llm_provider, encoding, logging_obj, optional_params, acompletion, timeout, vertex_project, vertex_location, vertex_credentials, gemini_api_key, litellm_params, logger_fn, extra_headers, client, api_base)
   1279 try:
-> 1280     response = client.post(url=url, headers=headers, json=data)  # type: ignore
   1281     response.raise_for_status()

File .venv/lib/python3.11/site-packages/litellm/llms/custom_httpx/http_handler.py:371, in HTTPHandler.post(self, url, data, json, params, headers, stream, timeout)
    370         setattr(e, "message", e.response.text)
--> 371     raise e
    372 except Exception as e:

File .venv/lib/python3.11/site-packages/litellm/llms/custom_httpx/http_handler.py:357, in HTTPHandler.post(self, url, data, json, params, headers, stream, timeout)
    356 response = self.client.send(req, stream=stream)
--> 357 response.raise_for_status()
    358 return response

File .venv/lib/python3.11/site-packages/httpx/_models.py:763, in Response.raise_for_status(self)
    762 message = message.format(self, error_type=error_type)
--> 763 raise HTTPStatusError(message, request=request, response=self)

HTTPStatusError: Client error '400 Bad Request' for url 'https://us-central1-aiplatform.googleapis.com/v1/projects/xxxxxx/locations/us-central1/publishers/google/models/gemini-1.5-pro:generateContent'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400

During handling of the above exception, another exception occurred:

VertexAIError                             Traceback (most recent call last)
File .venv/lib/python3.11/site-packages/litellm/main.py:2223, in completion(model, messages, timeout, temperature, top_p, n, stream, stream_options, stop, max_completion_tokens, max_tokens, presence_penalty, frequency_penalty, logit_bias, user, response_format, seed, tools, tool_choice, logprobs, top_logprobs, parallel_tool_calls, deployment_id, extra_headers, functions, function_call, base_url, api_version, api_key, model_list, **kwargs)
   2219 elif "gemini" in model or (
   2220     litellm_params.get("base_model") is not None
   2221     and "gemini" in litellm_params["base_model"]
   2222 ):
-> 2223     model_response = vertex_chat_completion.completion(  # type: ignore
   2224         model=model,
   2225         messages=messages,
   2226         model_response=model_response,
   2227         print_verbose=print_verbose,
   2228         optional_params=new_params,
   2229         litellm_params=litellm_params,  # type: ignore
   2230         logger_fn=logger_fn,
   2231         encoding=encoding,
   2232         vertex_location=vertex_ai_location,
   2233         vertex_project=vertex_ai_project,
   2234         vertex_credentials=vertex_credentials,
   2235         gemini_api_key=None,
   2236         logging_obj=logging,
   2237         acompletion=acompletion,
   2238         timeout=timeout,
   2239         custom_llm_provider=custom_llm_provider,
   2240         client=client,
   2241         api_base=api_base,
   2242         extra_headers=extra_headers,
   2243     )
   2244 else:

File .venv/lib/python3.11/site-packages/litellm/llms/vertex_ai_and_google_ai_studio/gemini/vertex_and_google_ai_studio_gemini.py:1284, in VertexLLM.completion(self, model, messages, model_response, print_verbose, custom_llm_provider, encoding, logging_obj, optional_params, acompletion, timeout, vertex_project, vertex_location, vertex_credentials, gemini_api_key, litellm_params, logger_fn, extra_headers, client, api_base)
   1283     error_code = err.response.status_code
-> 1284     raise VertexAIError(status_code=error_code, message=err.response.text)
   1285 except httpx.TimeoutException:

VertexAIError: {
  "error": {
    "code": 400,
    "message": "Invalid JSON payload received. Unknown name \"$defs\" at 'generation_config.response_schema': Cannot find field.\nInvalid JSON payload received. Unknown name \"$ref\" at 'generation_config.response_schema.properties[0].value.items': Cannot find field.",
    "status": "INVALID_ARGUMENT",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.BadRequest",
        "fieldViolations": [
          {
            "field": "generation_config.response_schema",
            "description": "Invalid JSON payload received. Unknown name \"$defs\" at 'generation_config.response_schema': Cannot find field."
          },
          {
            "field": "generation_config.response_schema.properties[0].value.items",
            "description": "Invalid JSON payload received. Unknown name \"$ref\" at 'generation_config.response_schema.properties[0].value.items': Cannot find field."
          }
        ]
      }
    ]
  }
}


During handling of the above exception, another exception occurred:

BadRequestError                           Traceback (most recent call last)
Cell In[184], line 4
      1 litellm.enable_json_schema_validation = True
      2 litellm.set_verbose = False # see the raw request made by litellm
----> 4 response = litellm.completion(
      5     model="vertex_ai/gemini-1.5-pro",
      6     messages=messages,
     14     response_format=EventsList
     15 )
     17 json.loads(response.choices[0].message.content)

File .venv/lib/python3.11/site-packages/litellm/utils.py:1063, in client.<locals>.wrapper(*args, **kwargs)
   1059         if (
   1060             liteDebuggerClient and liteDebuggerClient.dashboard_url != None
   1061         ):  # make it easy to get to the debugger logs if you've initialized it
   1062             e.message += f"\n Check the log in your dashboard - {liteDebuggerClient.dashboard_url}"
-> 1063 raise e

File .venv/lib/python3.11/site-packages/litellm/utils.py:947, in client.<locals>.wrapper(*args, **kwargs)
    945         print_verbose(f"Error while checking max token limit: {str(e)}")
    946 # MODEL CALL
--> 947 result = original_function(*args, **kwargs)
    948 end_time = datetime.datetime.now()
    949 if "stream" in kwargs and kwargs["stream"] is True:

File .venv/lib/python3.11/site-packages/litellm/main.py:2900, in completion(model, messages, timeout, temperature, top_p, n, stream, stream_options, stop, max_completion_tokens, max_tokens, presence_penalty, frequency_penalty, logit_bias, user, response_format, seed, tools, tool_choice, logprobs, top_logprobs, parallel_tool_calls, deployment_id, extra_headers, functions, function_call, base_url, api_version, api_key, model_list, **kwargs)
   2897     return response
   2898 except Exception as e:
   2899     ## Map to OpenAI Exception
-> 2900     raise exception_type(
   2901         model=model,
   2902         custom_llm_provider=custom_llm_provider,
   2903         original_exception=e,
   2904         completion_kwargs=args,
   2905         extra_kwargs=kwargs,
   2906     )

File .venv/lib/python3.11/site-packages/litellm/litellm_core_utils/exception_mapping_utils.py:2033, in exception_type(model, original_exception, custom_llm_provider, completion_kwargs, extra_kwargs)
   2031 if exception_mapping_worked:
   2032     setattr(e, "litellm_response_headers", litellm_response_headers)
-> 2033     raise e
   2034 else:
   2035     for error_type in litellm.LITELLM_EXCEPTION_TYPES:

File .venv/lib/python3.11/site-packages/litellm/litellm_core_utils/exception_mapping_utils.py:1049, in exception_type(model, original_exception, custom_llm_provider, completion_kwargs, extra_kwargs)
   1047 if original_exception.status_code == 400:
   1048     exception_mapping_worked = True
-> 1049     raise BadRequestError(
   1050         message=f"VertexAIException BadRequestError - {error_str}",
   1051         model=model,
   1052         llm_provider="vertex_ai",
   1053         litellm_debug_info=extra_information,
   1054         response=httpx.Response(
   1055             status_code=400,
   1056             request=httpx.Request(
   1057                 method="POST",
   1058                 url="https://cloud.google.com/vertex-ai/",
   1059             ),
   1060         ),
   1061     )
   1062 if original_exception.status_code == 401:
   1063     exception_mapping_worked = True

BadRequestError: litellm.BadRequestError: VertexAIException BadRequestError - {
  "error": {
    "code": 400,
    "message": "Invalid JSON payload received. Unknown name \"$defs\" at 'generation_config.response_schema': Cannot find field.\nInvalid JSON payload received. Unknown name \"$ref\" at 'generation_config.response_schema.properties[0].value.items': Cannot find field.",
    "status": "INVALID_ARGUMENT",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.BadRequest",
        "fieldViolations": [
          {
            "field": "generation_config.response_schema",
            "description": "Invalid JSON payload received. Unknown name \"$defs\" at 'generation_config.response_schema': Cannot find field."
          },
          {
            "field": "generation_config.response_schema.properties[0].value.items",
            "description": "Invalid JSON payload received. Unknown name \"$ref\" at 'generation_config.response_schema.properties[0].value.items': Cannot find field."
          }
        ]
      }
    ]
  }
}
```

**Comment by pvachon-ivadolabs** [comment]
After further testing, I can confirm that nested Pydantic classes work in the Google GenAI SDK, but not in LiteLLM.

Using LiteLLM
```python
messages = [{"role": "user", "content": "List important events from the 20th century."}]

class CalendarEvent(BaseModel):
      name: str
      date: str
      participants: list[str]

class EventsList(BaseModel):
    events: list[CalendarEvent]

response = litellm.completion(
    model="vertex_ai/gemini-1.5-flash",
    messages=messages,
    response_format=EventsList
)
```

Errors out as above. However, using the Google GenAI SDK:

```python
class CalendarEvent(BaseModel):
      name: str
      date: str
      participants: list[str]

class EventsList(BaseModel):
    events: list[CalendarEvent]

model = genai.GenerativeModel("gemini-1.5-flash")
result = model.generate_content(
    "List important events from the 20th century.",
    generation_config=genai.GenerationConfig(
        response_mime_type="application/json", response_schema=EventsList
    ),
)

print(result)
```

Returns OK
```
response:
GenerateContentResponse(
    done=True,
    iterator=None,
    result=protos.GenerateContentResponse({
      "candidates": [
        {
          "content": {
            "parts": [
              {
                "text": "{\"events\": [{\"date\": \"1903\", \"name\": \"Wright brothers first flight\", \"participants\": [\"Orville Wright\", \"Wilbur Wright\"]}, {\"date\": \"1914-1918\", \"name\": \"World War I\", \"participants\": [\"Austria-Hungary\", \"Germany\", \"Ottoman Empire\", \"Bulgaria\", \"France\", \"Russia\", \"Great Britain\", \"Italy\", \"United States\", \"Japan\"]}, {\"date\": \"1917\", \"name\": \"Russian Revolution\", \"participants\": [\"Bolsheviks\", \"Mensheviks\", \"White Russians\"]}, {\"date\": \"1929\", \"name\": \"Wall Street Crash\", \"participants\": [\"United States\"]}, {\"date\": \"1933\", \"name\": \"Adolf Hitler becomes Chancellor of Germany\", \"participants\": [\"Adolf Hitler\", \"Nazi Party\"]}, {\"date\": \"1939-1945\", \"name\": \"World War II\", \"participants\": [\"Germany\", \"Italy\", \"Japan\", \"Great Britain\", \"France\", \"Soviet Union\", \"United States\"]}, {\"date\": \"1945\", \"name\": \"Atomic bombings of Hiroshima and Nagasaki\", \"participants\": [\"United States\", \"Japan\"]}, {\"date\": \"1947\", \"name\": \"Truman Doctrine\", \"participants\": [\"United States\"]}, {\"date\": \"1948\", \"name\": \"Formation of NATO\", \"participants\": [\"United States\", \"Canada\", \"Belgium\", \"Denmark\", \"France\", \"Iceland\", \"Italy\", \"Luxembourg\", \"Netherlands\", \"Norway\", \"Portugal\", \"United Kingdom\"]}, {\"date\": \"1949\", \"name\": \"Formation of the People's Republic of China\", \"participants\": [\"Communist Party of China\"]}, {\"date\": \"1955\", \"name\": \"Warsaw Pact\", \"participants\": [\"Soviet Union\", \"Albania\", \"Bulgaria\", \"Czechoslovakia\", \"East Germany\", \"Hungary\", \"Poland\", \"Romania\"]}, {\"date\": \"1961\", \"name\": \"Construction of the Berlin Wall\", \"participants\": [\"Soviet Union\", \"East Germany\"]}, {\"date\": \"1962\", \"name\": \"Cuban Missile Crisis\", \"participants\": [\"United States\", \"Soviet Union\", \"Cuba\"]}, {\"date\": \"1968\", \"name\": \"Prague Spring\", \"participants\": [\"Czechoslovakia\", \"Soviet Union\"]}, {\"date\": \"1969\", \"name\": \"Apollo 11 landing on the Moon\", \"participants\": [\"United States\"]}, {\"date\": \"1973\", \"name\": \"Watergate scandal\", \"participants\": [\"United States\"]}, {\"date\": \"1979\", \"name\": \"Iranian Revolution\", \"participants\": [\"Iran\"]}, {\"date\": \"1989\", \"name\": \"Fall of the Berlin Wall\", \"participants\": [\"East Germany\", \"West Germany\"]}, {\"date\": \"1991\", \"name\": \"Dissolution of the Soviet Union\", \"participants\": [\"Soviet Union\"]}, {\"date\": \"2001\", \"name\": \"September 11 attacks\", \"participants\": [\"United States\"]}]}\n"
              }
            ],
            "role": "model"
          },
          "finish_reason": "STOP",
          "index": 0,
          "safety_ratings": [
            {
              "category": "HARM_CATEGORY_SEXUALLY_EXPLICIT",
              "probability": "NEGLIGIBLE"
            },
            {
              "category": "HARM_CATEGORY_HATE_SPEECH",
              "probability": "NEGLIGIBLE"
            },
            {
              "category": "HARM_CATEGORY_HARASSMENT",
              "probability": "NEGLIGIBLE"
            },
            {
              "category": "HARM_CATEGORY_DANGEROUS_CONTENT",
              "probability": "LOW"
            }
          ]
        }
      ],
      "usage_metadata": {
        "prompt_token_count": 12,
        "candidates_token_count": 693,
        "total_token_count": 705
      }
    }),
)
```

**Comment by ghost** [comment]
Thanks for the issue + debug @pvachon-ivadolabs 

Fix pushed https://github.com/BerriAI/litellm/commit/6f322c892fe4c0e976008a9716a46d827cb9ae79

Vertex has some quirks with what they accept in their json schema. thanks for the catch!

**Comment by pvachon-ivadolabs** [comment]
Oh awesome @krrishdholakia ! I'll test it on my end on Monday.

**Comment by ghost** [comment]
Curious @pvachon-ivadolabs how do you use litellm today? 

Trying to learn so we can improve for your use-case!

**Comment by pvachon-ivadolabs** [comment]
Tested it this morning on my end and it works! Thanks again! 🍾

**Comment by cebasfu93** [comment]
This is fixed for `gemini-1.5-pro`, but not all other versions.
I tested the following:

🔴 gemini-2.0-flash-lite-001"  # BAD
🟢 gemini-2.0-flash-001"  # GOOD
🟢 gemini-1.5-pro"  # GOOD
🔴 gemini-2.5-pro-preview-03-25"  # BAD
🟢 gemini-2.5-flash-preview-04-17"  # GOOD

**Comment by ghost** [comment]
Hey @cebasfu93 what's the error you see on gemini-2
